packages/cli/src/lib/setup/setupInstall.ts

```diff
@@ -947,34 +988,30 @@ function Install(options) {
                 : new RegExp(`^system\\.adapter\\.${adapter}\\.\\d+$`);
 
         try {
-            const doc = await objects.getObjectView('system', 'instance', {
+            const doc = await this.objects.getObjectView('system', 'instance', {
                 startkey: `system.adapter.${adapter}${instance !== undefined ? `.${instance}` : ''}`,
                 endkey: `system.adapter.${adapter}${instance !== undefined ? `.${instance}` : ''}\u9999`
             });
 
             // add non-duplicates to the list (if instance not given -> only for this host)
             const newObjIDs = doc.rows
                 // only the ones with an ID that matches the pattern
-                .filter(row => row && row.value && row.value._id)
-                .filter(row => instanceRegex.test(row.value._id))
+                .filter(row => row?.value && row.value._id)
+                .filter(row => row?.value && instanceRegex.test(row.value._id))
                 // if instance given also delete from foreign host else only instance on this host
                 .filter(row => {
-                    if (
-                        instance !== undefined ||
-                        !row.value.common ||
-                        !row.value.common.host ||
-                        row.value.common.host === hostname
-                    ) {
+                    if (instance !== undefined || row?.value?.common.host === hostname) {
                         return true;
                     } else {
-                        if (!notDeleted.includes(row.value._id)) {
+                        if (row.value && !notDeleted.includes(row.value._id)) {
                             notDeleted.push(row.value._id);
                         }
                         return false;
                     }
                 })
-                .map(row => row.value._id)
+                .map(row => row.value!._id)
                 .filter(id => !knownObjIDs.includes(id));
+            // eslint-disable-next-line prefer-spread
             knownObjIDs.push.apply(knownObjIDs, newObjIDs);
```

Comment:
Why not do what eslint suggests?
```suggestion
            knownObjIDs.push(...newObjIDs);
```

Reply:
The whole thing is, I did not want to touch it. WTF we are even using `apply` here?

Reply:
Ah okay.. just a hack to get around the array structure of newObjIds I guess

Reply:
The previous version was the "oldschool" way to push multiple array items into a different array. Since parameter spreads are available, this is no longer necessary, so eslint warns correctly.